### PR TITLE
Forkgc optimizations

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

**Comment by K-Jo** [comment]
@MeirShpilraien missing doc

**Comment by MeirShpilraien** [comment]
@K-Jo I added documentation. Can you pls take a look?

File: docs/Configuring.md [hunk 1]
```diff
@@ -232,11 +236,15 @@ interval (in seconds) in which to run the fork gc (relevant only when fork gc is
 $ redis-server --loadmodule ./redisearch.so GC_POLICY FORK FORK_GC_RUN_INTERVAL 60
 ```
 
+### Notes
+
+* only to be combined with `GC_POLICY FORK`
+* added in v1.4.16
+* not to be combined with `FORK_GC_CLEAN_THRESHOLD`
```
Comment (MeirShpilraien) [inline docs/Configuring.md]: Why no?

File: docs/Configuring.md [hunk 2]
```diff
@@ -248,10 +256,15 @@ fork to be created at the same time.
 $ redis-server --loadmodule ./redisearch.so GC_POLICY FORK FORK_GC_RETRY_INTERVAL 10
 ```
 
+### Notes
+
+* only to be combined with `GC_POLICY FORK` and `FORK_GC_RUN_INTERVAL`
+* added in v1.4.16
+* not to be combined with `FORK_GC_CLEAN_THRESHOLD`
```
Comment (MeirShpilraien) [inline docs/Configuring.md]: Why not?

File: docs/Configuring.md [hunk 3]
```diff
@@ -262,3 +275,9 @@ for behavior backword compatability but its highly recomended to change it to so
 ```
 $ redis-server --loadmodule ./redisearch.so GC_POLICY FORK FORK_GC_CLEAN_THRESHOLD 10000
 ```
+
+### Notes
+
+* only to be combined with `GC_POLICY FORK`
+* added in v1.4.16
+* not to be combined with `FORK_GC_RETRY_INTERVAL`
```
Comment (MeirShpilraien) [inline docs/Configuring.md]: Why not?

**Comment by MeirShpilraien** [comment]
@mnunberg I made some changes, can you take another look?

File: docs/Configuring.md [hunk 4]
```diff
@@ -217,3 +217,63 @@ The policy for the garbage collector. Supported policies are:
...
+$ redis-server --loadmodule ./redisearch.so GC_POLICY FORK FORK_GC_RUN_INTERVAL 60
+```
+
+### Notes
+
+* only to be combined with `GC_POLICY FORK`
+
+## FORK_GC_RETRY_INTERVAL
+
+Interval (in seconds) in which RediSearch will retry to run `fork GC` in case of a failure. Usually, a failure could happen when the redis fork api does not allow for more then one fork to be created at the same time.
```
Comment (mnunberg) [inline docs/Configuring.md]: more *than*

**Comment by MeirShpilraien** [comment]
@mnunberg yes I will cherrypick after we merge

File: src/config.h [hunk 5]
```diff
@@ -76,6 +76,7 @@ typedef struct {
 
   GCPolicy gcPolicy;
   size_t forkGcRunIntervalSec;
+  size_t forkGcCleanThreashold;
```
Comment (mnunberg) [inline src/config.h]: `threshold`, not `threashold`

**Pull request merged**